src/BTCPayServer.Lightning.LND/LndSwaggerClient.partial.cs

```diff
@@ -117,6 +118,16 @@ internal static HttpClient CreateHttpClient(LndRestSettings settings, HttpClient
                     var hash = GetHash(actualCert);
                     return hash.SequenceEqual(expectedThumbprint);
                 };
+            } else if (settings.CertificateFilePath != null) {
+                handler.ServerCertificateCustomValidationCallback = (request, cert, chain, errors) =>
+                {
+                    if (cert == null) throw new ArgumentNullException("cert");
+                    var expectedCollection = new X509Certificate2Collection();
+                    expectedCollection.ImportFromPemFile(settings.CertificateFilePath);
+                    if (!expectedCollection.Contains(cert))
```

Comment:
It shouldn't throw, it should return false.

Reply:
I figured the failure would be more understandable with an explicit message. Helped me a lot with debugging.